### PR TITLE
Thesilence docs

### PR DESCRIPTION
**Pull request by therealsilence**
- Add docs on automation (triggers / cron / macros / stub for dmon).
- Move examples from Storm command reference to automation doc.
- Add docs on subqueries / subquery filters.
- Additional subquery filter example.
- Add docs on Storm ps.* commands; note equivalent cmdr commands are deprecated.
- Add section on autoadd and depadd in data mod reference.
- Add "pivot in and walk" to go with "pivot out and walk" in pivot reference docs.
- Update type-specific docs for arrays to include "edit try" syntax.
- Cross-reference updates, glossary updates, minor fixes / typos.

### CODE REVIEW

File: docs/synapse/userguides/storm_ref_automation.ipynb [hunk 1]
```diff
@@ -11,30 +43,675 @@
...
+    "\n",
+    "Background\n",
+    "----------\n",
+    "\n",
+    "Synapse is designed to facilitate large-scale analysis over disparate data sources with speed and efficiency. Many of the features that support this analysis are built into Synapse’s architecture, from performance-optimized indexing and storage to an extensible data model that allows you to reason over data in a structured manner.\n",
+    "\n",
+    "Synapse also supports large-scale analysis through the use of **automation.** Synapse’s automation features are available through the Storm runtime and include:\n",
+    "\n",
+    "- `Triggers and Cron`_\n",
+    "- `Macros`_\n",
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_automation.ipynb:56]: Storm Dmons need to be listed here, even if we leave a sectino for them marked as TDB

File: docs/synapse/userguides/storm_ref_automation.ipynb [hunk 2]
```diff
@@ -11,30 +43,675 @@
...
+    ".. _auto-bkgd:\n",
+    "\n",
+    "Background\n",
+    "----------\n",
+    "\n",
+    "Synapse is designed to facilitate large-scale analysis over disparate data sources with speed and efficiency. Many of the features that support this analysis are built into Synapse’s architecture, from performance-optimized indexing and storage to an extensible data model that allows you to reason over data in a structured manner.\n",
+    "\n",
+    "Synapse also supports large-scale analysis through the use of **automation.** Synapse’s automation features are available through the Storm runtime and include:\n",
+    "\n",
+    "- `Triggers and Cron`_\n",
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_automation.ipynb:55]: Triggers and Cron are separate line items

File: docs/synapse/userguides/storm_ref_automation.ipynb [hunk 3]
```diff
@@ -11,30 +43,675 @@
...
+    "- `Macros`_\n",
+    "\n",
+    "Triggers, cron, and macros in Synapse all utilize the Storm query language. That is, regardless of the type of automation used, the result is the execution of a predefined Storm query. This means that **anything that can be written in Storm can be automated,** from the simple and straightforward to the highly complex. Actions performed via automation are limited only by imagination and Storm proficiency. Some automation is fairly simple (\"if X occurs, do Y\" or \"once a week, update Z\"). However, automation can take advantage of all available Storm features, including subqueries, variables, libraries, control flow logic, and so on.\n",
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_automation.ipynb:58]: dmons need to be listed.

File: docs/synapse/userguides/storm_ref_automation.ipynb [hunk 4]
```diff
@@ -11,30 +43,675 @@
...
+    "\n",
+    "This section is not meant to be a detailed guide on implementing automation. However, a few areas are included here for consideration when utilizing automation in your environment.\n",
+    "\n",
+    "**Permissions.**\n",
+    "\n",
+    "The ability to create, modify, or delete triggers, cron jobs, and macros depends on a user having the appropriate permissions to do so. If you assign permissions for these tasks in your environment, keep the following in mind:\n",
+    "\n",
+    "- Triggers and cron jobs execute in the context of (with the permissions of) **the user who creates them.**\n",
+    "\n",
+    "- Macros execute in the context of **the user who calls them** for execution, regardless of who created them.\n",
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_automation.ipynb:73]: Dmons execute in the context of the user that created it, in the view it was created in.

File: docs/synapse/userguides/storm_ref_automation.ipynb [hunk 5]
```diff
@@ -11,30 +43,675 @@
...
+    "\n",
+    "The Storm query executed by a given automation task (whether trigger, cron, or macro) cannot perform tasks that it does not have appropriate permissions to perform. This leads to some straightforward (and some less straightforward) situations. For example:\n",
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_automation.ipynb:75]: No need to enumerate the automation styles here

File: docs/synapse/userguides/storm_ref_automation.ipynb [hunk 6]
```diff
@@ -11,30 +43,675 @@
...
+    "\n",
+    "- an analyst may write a cron job to execute a long-running data collection and analysis task during off-peak hours, or may create a macro as a \"shortcut\" to make it easier to store and execute a particular Storm query on demand.\n",
+    "\n",
+    "- an organization may implement a set of triggers, cron jobs, and macros to track malicious indicators within Synapse and orchestrate the integration of those indicators with security devices or SIEM systems.\n",
+    "\n",
+    "**Architecture.**\n",
+    "\n",
+    "There are varying approaches for \"how\" to write and implement automation. For example:\n",
+    "\n",
+    "- Triggers, cron, and macros can be kept entirely separate and independent from one another, each executing their own Storm code and performing different tasks. Alternately, automation can be somewhat centralized within macros, where triggers and cron (when used) execute minimal Storm queries whose purpose is to call more extensive Storm stored in macros.\n",
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_automation.ipynb:93]: another places where everything is listed together

File: docs/synapse/userguides/storm_ref_automation.ipynb [hunk 7]
```diff
@@ -11,30 +43,675 @@
...
+    "podes = await core.eval(q2, num=1, cmdr=False)\n",
+    "podes = await core.eval(q3, num=1, cmdr=False)"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    ".. CAUTION::\n",
+    "  This example illustrates a potential \"order of operations\" issue as described in the background section above. When the ``inet:dns:request`` node is created, if the ``file:bytes`` node referenced by the ``:exe`` secondary property **does not** already exist, Synapse will create it automatically as a \"derivative node\". However, the trigger will fire as soon as the ``:exe`` property is set - which occurs before the ``file:bytes`` node is created. In this case, the file will **not** be tagged ``#review`` because it does not yet exist when the trigger attempts to apply the tag. (In contrast, if the ``file:bytes`` node was already present in the cortex, it would be tagged as expected.)"
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_automation.ipynb:343]: I thought we were resolving autoadds before triggers were executed?

File: docs/synapse/userguides/storm_ref_automation.ipynb [hunk 8]
```diff
@@ -11,30 +43,675 @@
...
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    ".. NOTE::\n",
+    "  The macro above uses variables, methods, and a switch case for control flow."
+   ]
+  },
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_automation.ipynb:698]: add a todo - stormdmons section

File: docs/synapse/userguides/storm_ref_automation.ipynb (hunk 2 above)
Comment (therealsilence) [reply docs/synapse/userguides/storm_ref_automation.ipynb:55]: They share an intro (and thus a single hyperlink) due to some similarities (from a user perspective) that are useful to discuss together. They each have their own subsection under that intro.

**Pull request merged**